Formula/a52dec.rb

```diff
@@ -28,4 +28,9 @@ def install
     touch testpath/"test"
     system "#{bin}/a52dec", "-o", "null", "test"
   end
+
+  livecheck do
+    url "http://liba52.sourceforge.net"
```

Comment:
This is also the homepage, can't livecheck infer this?

Reply:
The homepage ends in `.io`, and while it redirects to the `.net` URL, livecheck will error on redirection.

When I remove the URL from the livecheckable:
```
Error: a52dec: redirection forbidden: https://liba52.sourceforge.io/files/a52dec-0.7.4.tar.gz -> http://liba52.sourceforge.net/files/a52dec-0.7.4.tar.gz
```

Reply:
> The homepage ends in `.io`, and while it redirects to the `.net` URL, livecheck will error on redirection.

We should fix this 👍

Reply:
While I was tinkering around my local livecheck  repository, I tested out using the Formula's URLs (`url`, `homepage`) rather than specifying `http://liba52.sourceforge.net` in the block.

Using the `--debug` option (and ensuring that the block was being used), I saw that `version_heuristic` errors as it first uses the `url` from Formula before the `homepage`. I'll see if there's a fix for this.

Reply:
> We should fix this 👍

@MikeMcQuaid This is the `sourceforge.io` issue that I previously detailed in Slack. As mentioned there, we can't update the homepage to `http://liba52.sourceforge.net` without the related audit failing, which is why I proposed removing/disabling that particular audit.


> While I was tinkering around my local livecheck repository, I tested out using the Formula's URLs (`url`, `homepage`) rather than specifying `http://liba52.sourceforge.net` in the block.

It's possible to use `self.homepage` (or `self.head.url`, `self.stable.url`) as the URL in a livecheckable but we've avoided it up to now.

Being able to specify the particular URL to use, from the ones that livecheck already uses (head, stable, homepage), has been on my to-do list for a bit. I was tentatively thinking that we could just provide a symbol like `:head`, `:stable`, or `:homepage` and livecheck would use that argument to get the associated URL behind the scenes. This way, people wouldn't have to remember how to access these URLs using `self` (and if `.url` is needed).

It's very simple to implement, I just haven't gotten to it yet. I can throw this together probably sometime today if it's desirable, though.

Reply:
I'd rather we figure out a way to do all this implicitly rather than explicitly. There's no technical reason the livecheck URL can't be parsed from the homepage and/or formula URL.

Reply:
> I'd rather we figure out a way to do all this implicitly rather than explicitly. There's no technical reason the livecheck URL can't be parsed from the homepage and/or formula URL.

I'm not sure I fully understand. What does "explicit" or "implicit" look like in this context? For example, would the `url :homepage` setup be counted as explicit?

That said, I feel like we're maybe getting a little in the weeds on what this particular livecheckable is doing or how livecheck currently behaves. It's incredibly unfortunate that this particular livecheckable was selected (due to being first alphabetically), as it's part of the 1% of existing livecheckables that don't work properly (we have 1000+ livecheckables at present). All existing livecheckables worked properly as of early March (we put a lot of time/effort into making that so) but a few have naturally broken since then and need to be updated.

There are no shortage of ways that livecheck and the current livecheckables could be improved if/when a feature is added or we get the time to bring old livecheckables up to current standards (primarily related to older, looser regexes). If we want to improve livecheck and/or livecheckables before migration, it's probably more appropriate to discuss and make changes in Homebrew/homebrew-livecheck, through topical issues and/or PRs.

Reply:
> I'm not sure I fully understand. What does "explicit" or "implicit" look like in this context? For example, would the `url :homepage` setup be counted as explicit?

Sorry, I should have been more explicit. I mean "without manually specifying a URL that looks very similar to/redirects from the `homepage` or `url`".

Reply:
> I mean "without manually specifying a URL that looks very similar to/redirects from the `homepage` or `url`".

Gotcha, this particular issue is already on my radar and will be improved through various planned feature additions, primarily `using`/`not_using` and allowing URLs from the formula to be easily specified (potentially something like `url :homepage`). I'll solicit more feedback when I'm working on each.

Ideally we would only use hardcoded URLs in livecheckables when it's really appropriate but right now it happens more often than I would like due to the current technical constraints of livecheck/livecheckables.

For example, we have some livecheckables that hardcode the formula's `head` URL to 1) ensure we're using a specific URL from the formula and 2) ensure we're using a specific version-finding strategy in the heuristic. The aforementioned features address those issues, so we'll be able to get rid of unnecessary duplication in the near future.